cql3/statements/alter_keyspace_statement.cc

```diff
@@ -50,6 +50,13 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
         }
         try {
             auto ks = qp.db().find_keyspace(_name);
+            // TODO: the full tablets support will be implemented in https://github.com/scylladb/scylladb/issues/16129
+            if (ks.get_replication_strategy().uses_tablets() && !_attrs->get_initial_tablets(ks.metadata()->strategy_name(), true).has_value()) {
+                throw exceptions::invalid_request_exception("Disabling tablets via ALTER KEYSPACE statement is not yet supported for keyspaces with tablets.");
+            }
+            if (ks.get_replication_strategy().uses_tablets() && !_attrs->get_replication_options().empty()) {
+                throw exceptions::invalid_request_exception("Changing replication options via ALTER KEYSPACE statement is not yet supported for keyspaces with tablets.");
+            }
```

Comment:
> I assume that recently promoted #18899 is prerequisite to this restriction, isn't it? Maybe it's worth moving this one lower, after the

Yes. I didn't know how to proceed, and since #18899 is in some limbo state, I wrote my code assuming it's not there yet (and it may be that it's never going to be merged for some reason?) and I need to protect against switching between vnodes/tablets.

Reply:
Now I remember it was also because of https://github.com/scylladb/scylladb/pull/18772#issuecomment-2124320898.
But since the commit with the check is already implemented by you, and just waits to finally arrive to the branch, I'll remove my redundant check.

Reply:
Removed the check for switching from tablets to vnodes.

Reply:
@ptrsmrn , you put :+1: and resolved the issue, but the code remained as it was before my suggestion, so I'm totally confused

Reply:
@xemul sorry, I just want to move quick, maybe I'm too quick and cut corners:

> I assume that recently promoted https://github.com/scylladb/scylladb/pull/18899 is prerequisite to this restriction, isn't it?

Yes, it's a prerequisite. 

As for the 2nd part of your comment, I assumed it's only about refactoring, and since I didn't want to refactor the code, build, test, push, re-run the CI (not to waste time), I ignored it.